All requests blocked by cloudflare (403 ERROR: The request could not be satisfied)

Starting today morning at `[2021-10-05 04:48:05]` UTC i first saw this error message and the problem still exists. 

Quick play around with the user agent did not result in anything better. 

```python
Python 3.9.1 (v3.9.1:1e5d33e9b9, Dec  7 2020, 12:10:52) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tgtg
>>> client = tgtg.TgtgClient(email='REDACTED', password='REDACTED')
>>> client.get_items()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/winterstein/venv/togoodtogo/lib/python3.9/site-packages/tgtg/__init__.py", line 143, in get_items
    self._login()
  File "/Users/winterstein/venv/togoodtogo/lib/python3.9/site-packages/tgtg/__init__.py", line 122, in _login
    raise TgtgLoginError(response.status_code, response.content)
tgtg.exceptions.TgtgLoginError: (403, b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>403 ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size="1px">\nRequest blocked.\nWe can\'t connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.\n<BR clear="all">\nIf you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.\n<BR clear="all">\n<HR noshade size="1px">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: REDACTED==\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>')
>>> 
```

Can also be verified by running `make test` with env vars `TGTG_EMAIL` and `TGTG_PASSWORD`:
```python
TGTG_EMAIL=REDACTED TGTG_PASSWORD=REDACTED make test
python3 -m poetry run pytest
=========================================================================================================================================================== test session starts ============================================================================================================================================================
platform darwin -- Python 3.9.1, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /Users/winterstein/tmp/tgtg-python, configfile: setup.cfg
plugins: cov-2.12.1, responses-0.4.0
collected 19 items                                                                                                                                                                                                                                                                                                                         

tests/test_api.py FF                                                                                                                                                                                                                                                                                                                 [ 10%]
tests/test_items.py .......                                                                                                                                                                                                                                                                                                          [ 47%]
tests/test_login.py ........                                                                                                                                                                                                                                                                                                         [ 89%]
tests/test_signup.py ..                                                                                                                                                                                                                                                                                                              [100%]

================================================================================================================================================================= FAILURES =================================================================================================================================================================
_____________________________________________________________________________________________________________________________________________________ TestLoginRequired.test_get_items _____________________________________________________________________________________________________________________________________________________

self = <tests.test_api.TestLoginRequired object at 0x7f94a8e0ce80>

    def test_get_items(self):
        client = TgtgClient(
            email=os.environ["TGTG_EMAIL"], password=os.environ["TGTG_PASSWORD"]
        )
>       data = client.get_items(
            favorites_only=False, radius=10, latitude=48.126, longitude=-1.723
        )

tests/test_api.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tgtg/__init__.py:143: in get_items
    self._login()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tgtg.TgtgClient object at 0x7f94a8e0cfd0>

    def _login(self):
        if self._already_logged:
            self._refresh_token()
        else:
            if not self.email or not self.password:
                raise ValueError(
                    "You must fill email and password or access_token and user_id"
                )
    
            response = requests.post(
                self._get_url(LOGIN_ENDPOINT),
                headers=self._headers,
                json={
                    "device_type": "ANDROID",
                    "email": self.email,
                    "password": self.password,
                },
                proxies=self.proxies,
                timeout=self.timeout,
            )
            if response.status_code == HTTPStatus.OK:
                login_response = response.json()
                self.access_token = login_response["access_token"]
                self.refresh_token = login_response["refresh_token"]
                self.last_time_token_refreshed = datetime.datetime.now()
                self.user_id = login_response["startup_data"]["user"]["user_id"]
            else:
>               raise TgtgLoginError(response.status_code, response.content)
E               tgtg.exceptions.TgtgLoginError: (403, b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>403 ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size="1px">\nRequest blocked.\nWe can\'t connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.\n<BR clear="all">\nIf you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.\n<BR clear="all">\n<HR noshade size="1px">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: HIqmDLzs--EUvFrZjM1PFLMVxKjlP5H6tUe0xj0uJdLDxtgQZllfvA==\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>')

tgtg/__init__.py:122: TgtgLoginError
___________________________________________________________________________________________________________________________________________________ TestLoginRequired.test_get_one_item ____________________________________________________________________________________________________________________________________________________

self = <tests.test_api.TestLoginRequired object at 0x7f94889a6ee0>

    def test_get_one_item(self):
        client = TgtgClient(
            email=os.environ["TGTG_EMAIL"], password=os.environ["TGTG_PASSWORD"]
        )
        item_id = "36684"
>       data = client.get_item(item_id)

tests/test_api.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tgtg/__init__.py:176: in get_item
    self._login()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tgtg.TgtgClient object at 0x7f94889a6df0>

    def _login(self):
        if self._already_logged:
            self._refresh_token()
        else:
            if not self.email or not self.password:
                raise ValueError(
                    "You must fill email and password or access_token and user_id"
                )
    
            response = requests.post(
                self._get_url(LOGIN_ENDPOINT),
                headers=self._headers,
                json={
                    "device_type": "ANDROID",
                    "email": self.email,
                    "password": self.password,
                },
                proxies=self.proxies,
                timeout=self.timeout,
            )
            if response.status_code == HTTPStatus.OK:
                login_response = response.json()
                self.access_token = login_response["access_token"]
                self.refresh_token = login_response["refresh_token"]
                self.last_time_token_refreshed = datetime.datetime.now()
                self.user_id = login_response["startup_data"]["user"]["user_id"]
            else:
>               raise TgtgLoginError(response.status_code, response.content)
E               tgtg.exceptions.TgtgLoginError: (403, b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>403 ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size="1px">\nRequest blocked.\nWe can\'t connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.\n<BR clear="all">\nIf you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.\n<BR clear="all">\n<HR noshade size="1px">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: p2OsFEJcRChVWR_TEszPc3j9SB4JKnMP5uCCKl04g4BmpjJOKUZYxA==\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>')

tgtg/__init__.py:122: TgtgLoginError
============================================================================================================================================================= warnings summary =============================================================================================================================================================
tests/test_login.py::test_login_empty_token_fail
  /Users/winterstein/tmp/tgtg-python/tgtg/__init__.py:53: UserWarning: 'user_id' is deprecated; use 'email' and 'password'
    warnings.warn("'user_id' is deprecated; use 'email' and 'password'")

tests/test_login.py::test_login_empty_user_id_fail
  /Users/winterstein/tmp/tgtg-python/tgtg/__init__.py:46: UserWarning: 'access_token' is deprecated; use 'email' and 'password'
    warnings.warn("'access_token' is deprecated; use 'email' and 'password'")

-- Docs: https://docs.pytest.org/en/stable/warnings.html

---------- coverage: platform darwin, python 3.9.1-final-0 -----------
Name                 Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------
tgtg/__init__.py        93      0     24      0   100%
tgtg/exceptions.py       4      0      0      0   100%
----------------------------------------------------------------
TOTAL                   97      0     24      0   100%

========================================================================================================================================================= short test summary info ==========================================================================================================================================================
FAILED tests/test_api.py::TestLoginRequired::test_get_items - tgtg.exceptions.TgtgLoginError: (403, b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could not b...
FAILED tests/test_api.py::TestLoginRequired::test_get_one_item - tgtg.exceptions.TgtgLoginError: (403, b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could no...
================================================================================================================================================= 2 failed, 17 passed, 2 warnings in 1.22s =================================================================================================================================================
make: *** [test] Error 1
```

--- a/tgtg/__init__.py
+++ b/tgtg/__init__.py
@@ -15,9 +15,9 @@ SIGNUP_BY_EMAIL_ENDPOINT = "auth/v2/signUpByEmail"
 REFRESH_ENDPOINT = "auth/v1/token/refresh"
 ALL_BUSINESS_ENDPOINT = "map/v1/listAllBusinessMap"
 USER_AGENTS = [
-    "TGTG/20.12.3 Dalvik/2.1.0 (Linux; U; Android 6.0.1; Nexus 5 Build/M4B30Z)",
-    "TGTG/20.12.3 Dalvik/2.1.0 (Linux; U; Android 7.0; SM-G935F Build/NRD90M)",
-    "TGTG/20.12.3 Dalvik/2.1.0 (Linux; Android 6.0.1; SM-G920V Build/MMB29K)",
+    "TGTG/21.9.3 Dalvik/2.1.0 (Linux; U; Android 6.0.1; Nexus 5 Build/M4B30Z)",
+    "TGTG/21.9.3 Dalvik/2.1.0 (Linux; U; Android 7.0; SM-G935F Build/NRD90M)",
+    "TGTG/21.9.3 Dalvik/2.1.0 (Linux; Android 6.0.1; SM-G920V Build/MMB29K)",
 ]
 DEFAULT_ACCESS_TOKEN_LIFETIME = 3600 * 4  # 4 hours
 

